### DISCUSSION TITLE
Shareable Input Between Triggers & Actions

### DISCUSSION THREAD

**Discussion by hanifaudah** [bahasa-ai/otomatis-guidance#11] [Ideas]
Some triggers and/or actions for a certain app require the exact same input. It would be much more convenient if there is a way to create a general input that can be used for multiple triggers/actions or maybe a way to import inputs from other triggers/actions.

For example, these inputs are required for every Accurate trigger & action
![image](https://user-images.githubusercontent.com/56548631/128655725-241703eb-c09d-471f-b8a1-be1c025a5e98.png)

**Comment by mgilangjanuar** [comment]
Good idea, let us figure out about this

**Comment by jundialwan** [comment]
This is good idea I think. "copy field to trigger/action" is more desirable rather than one field referred/used by two action/trigger.

Shareable object is a headache. Two edge cases for shareable field:
- Deletion in origin action/trigger because field is not used anymore. How we handle that field on the triggers/action that refer to it?
- In case of dynamic field it is difficult to maintain two logic flow inside one dynamic form function. Keep Single-Responsibility Purpose always. Ref: https://en.wikipedia.org/wiki/Single-responsibility_principle
